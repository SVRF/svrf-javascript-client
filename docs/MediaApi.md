# SVRF.MediaApi

All URIs are relative to *https://api.svrf.com/v1*

Method | HTTP request | Description
------------- | ------------- | -------------
[**getById**](MediaApi.md#getById) | **GET** /vr/{id} | Media by ID Endpoint
[**getTrending**](MediaApi.md#getTrending) | **GET** /vr/trending | Trending Endpoint
[**search**](MediaApi.md#search) | **GET** /vr/search | Search Endpoint


<a name="getById"></a>
# **getById**
> SingleMediaResponse getById(id)

Media by ID Endpoint

Fetch media by its ID.

### Example
```javascript
var SVRF = require('svrf-client');
var defaultClient = SVRF.ApiClient.instance;

// Configure API key authorization: XAppToken
var XAppToken = defaultClient.authentications['XAppToken'];
XAppToken.apiKey = 'YOUR API KEY';
// Uncomment the following line to set a prefix for the API key, e.g. "Token" (defaults to null)
//XAppToken.apiKeyPrefix = 'Token';

var apiInstance = new SVRF.MediaApi();

var id = "id_example"; // String | ID of Media

apiInstance.getById(id).then(function(data) {
  console.log('API called successfully. Returned data: ' + data);
}, function(error) {
  console.error(error);
});

```

### Parameters

Name | Type | Description  | Notes
------------- | ------------- | ------------- | -------------
 **id** | **String**| ID of Media | 

### Return type

[**SingleMediaResponse**](SingleMediaResponse.md)

### Authorization

[XAppToken](../README.md#XAppToken)

### HTTP request headers

 - **Content-Type**: Not defined
 - **Accept**: application/json

<a name="getTrending"></a>
# **getTrending**
> TrendingResponse getTrending(opts)

Trending Endpoint

The SVRF Trending Endpoint provides your app or project with the hottest immersive content curated by real humans. The experiences returned mirror the [SVRF homepage](https://www.svrf.com), from timely cultural content to trending pop-culture references. The trending experiences are updated regularly to ensure users always get fresh updates of immersive content.

### Example
```javascript
var SVRF = require('svrf-client');
var defaultClient = SVRF.ApiClient.instance;

// Configure API key authorization: XAppToken
var XAppToken = defaultClient.authentications['XAppToken'];
XAppToken.apiKey = 'YOUR API KEY';
// Uncomment the following line to set a prefix for the API key, e.g. "Token" (defaults to null)
//XAppToken.apiKeyPrefix = 'Token';

var apiInstance = new SVRF.MediaApi();

var opts = { 
  'type': [new SVRF.MediaType()], // [MediaType] | The type(s) of Media to be returned (comma separated).
  'stereoscopicType': "stereoscopicType_example", // String | Search only for Media with a particular stereoscopic type.
  'category': "category_example", // String | Search only for Media with a particular category.
  'size': 10, // Number | The number of results per page.
  'minimumWidth': 56, // Number | The minimum width for video and photo Media, in pixels.
  'pageNum': 56 // Number | Pagination control to fetch the next page of results, if applicable.
};
apiInstance.getTrending(opts).then(function(data) {
  console.log('API called successfully. Returned data: ' + data);
}, function(error) {
  console.error(error);
});

```

### Parameters

Name | Type | Description  | Notes
------------- | ------------- | ------------- | -------------
 **type** | [**[MediaType]**](MediaType.md)| The type(s) of Media to be returned (comma separated). | [optional] 
 **stereoscopicType** | **String**| Search only for Media with a particular stereoscopic type. | [optional] 
 **category** | **String**| Search only for Media with a particular category. | [optional] 
 **size** | **Number**| The number of results per page. | [optional] [default to 10]
 **minimumWidth** | **Number**| The minimum width for video and photo Media, in pixels. | [optional] 
 **pageNum** | **Number**| Pagination control to fetch the next page of results, if applicable. | [optional] 

### Return type

[**TrendingResponse**](TrendingResponse.md)

### Authorization

[XAppToken](../README.md#XAppToken)

### HTTP request headers

 - **Content-Type**: Not defined
 - **Accept**: application/json

<a name="search"></a>
# **search**
> SearchMediaResponse search(q, opts)

Search Endpoint

The SVRF Search Endpoint brings the power of immersive search found on [SVRF.com](https://www.svrf.com) to your app or project. SVRF&#39;s search engine enables your users to instantly find the immersive experience they&#39;re seeking. Content is sorted by the SVRF rating system, ensuring that the highest quality content and most prevalent search results are returned. 

### Example
```javascript
var SVRF = require('svrf-client');
var defaultClient = SVRF.ApiClient.instance;

// Configure API key authorization: XAppToken
var XAppToken = defaultClient.authentications['XAppToken'];
XAppToken.apiKey = 'YOUR API KEY';
// Uncomment the following line to set a prefix for the API key, e.g. "Token" (defaults to null)
//XAppToken.apiKeyPrefix = 'Token';

var apiInstance = new SVRF.MediaApi();

var q = "q_example"; // String | Url-encoded search query.

var opts = { 
  'type': [new SVRF.MediaType()], // [MediaType] | The type(s) of Media to be returned (comma separated).
  'stereoscopicType': "stereoscopicType_example", // String | Search only for Media with a particular stereoscopic type.
  'category': "category_example", // String | Search only for Media with a particular category.
  'size': 10, // Number | The number of results to return per-page, from 1 to 100.
  'minimumWidth': 56, // Number | The minimum width for video and photo Media, in pixels.
  'pageNum': 56 // Number | Pagination control to fetch the next page of results, if applicable.
};
apiInstance.search(q, opts).then(function(data) {
  console.log('API called successfully. Returned data: ' + data);
}, function(error) {
  console.error(error);
});

```

### Parameters

Name | Type | Description  | Notes
------------- | ------------- | ------------- | -------------
 **q** | **String**| Url-encoded search query. | 
 **type** | [**[MediaType]**](MediaType.md)| The type(s) of Media to be returned (comma separated). | [optional] 
 **stereoscopicType** | **String**| Search only for Media with a particular stereoscopic type. | [optional] 
 **category** | **String**| Search only for Media with a particular category. | [optional] 
 **size** | **Number**| The number of results to return per-page, from 1 to 100. | [optional] [default to 10]
 **minimumWidth** | **Number**| The minimum width for video and photo Media, in pixels. | [optional] 
 **pageNum** | **Number**| Pagination control to fetch the next page of results, if applicable. | [optional] 

### Return type

[**SearchMediaResponse**](SearchMediaResponse.md)

### Authorization

[XAppToken](../README.md#XAppToken)

### HTTP request headers

 - **Content-Type**: Not defined
 - **Accept**: application/json

