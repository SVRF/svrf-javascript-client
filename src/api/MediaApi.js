/*
 * SVRF API
 *
 * OpenAPI spec version: 1.0.0
 * Contact: api@svrf.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/ErrorResponse', 'model/MediaType', 'model/RateLimitResponse', 'model/SearchMediaResponse', 'model/SingleMediaResponse', 'model/TrendingResponse'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('../model/ErrorResponse'), require('../model/MediaType'), require('../model/RateLimitResponse'), require('../model/SearchMediaResponse'), require('../model/SingleMediaResponse'), require('../model/TrendingResponse'));
  } else {
    // Browser globals (root is window)
    if (!root.SVRF) {
      root.SVRF = {};
    }
    root.SVRF.MediaApi = factory(root.SVRF.ApiClient, root.SVRF.ErrorResponse, root.SVRF.MediaType, root.SVRF.RateLimitResponse, root.SVRF.SearchMediaResponse, root.SVRF.SingleMediaResponse, root.SVRF.TrendingResponse);
  }
}(this, function(ApiClient, ErrorResponse, MediaType, RateLimitResponse, SearchMediaResponse, SingleMediaResponse, TrendingResponse) {
  'use strict';

  /**
   * Media service.
   * @module api/MediaApi
   * @version 1.5.0
   */

  /**
   * Constructs a new MediaApi. 
   * @alias module:api/MediaApi
   * @class
   * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
   * default to {@link module:ApiClient#instance} if unspecified.
   */
  var exports = function(apiClient) {
    this.apiClient = apiClient || ApiClient.instance;



    /**
     * Media by ID Endpoint
     * Fetch media by its ID.
     * @param {String} id ID of Media
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/SingleMediaResponse} and HTTP response
     */
    this.getByIdWithHttpInfo = function(id) {
      var postBody = null;

      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling getById");
      }


      var pathParams = {
        'id': id
      };
      var queryParams = {
      };
      var collectionQueryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = ['XAppToken'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = SingleMediaResponse;

      return this.apiClient.callApi(
        '/vr/{id}', 'GET',
        pathParams, queryParams, collectionQueryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType
      );
    }

    /**
     * Media by ID Endpoint
     * Fetch media by its ID.
     * @param {String} id ID of Media
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/SingleMediaResponse}
     */
    this.getById = function(id) {
      return this.getByIdWithHttpInfo(id)
        .then(function(response_and_data) {
          return response_and_data.data;
        });
    }


    /**
     * Trending Endpoint
     * The SVRF Trending Endpoint provides your app or project with the hottest immersive content curated by real humans. The experiences returned mirror the [SVRF homepage](https://www.svrf.com), from timely cultural content to trending pop-culture references. The trending experiences are updated regularly to ensure users always get fresh updates of immersive content.
     * @param {Object} opts Optional parameters
     * @param {Array.<module:model/MediaType>} opts.type The type(s) of Media to be returned (comma separated).
     * @param {module:model/String} opts.stereoscopicType Search only for Media with a particular stereoscopic type.
     * @param {module:model/String} opts.category Search only for Media with a particular category.
     * @param {Number} opts.size The number of results per page. (default to 10)
     * @param {Number} opts.minimumWidth The minimum width for video and photo Media, in pixels.
     * @param {Number} opts.pageNum Pagination control to fetch the next page of results, if applicable.
     * @param {Boolean} opts.isFaceFilter Search only for Face Filters.
     * @param {Boolean} opts.hasBlendShapes Search only for Media that has blend shapes.
     * @param {Boolean} opts.requiresBlendShapes Search only for Media that requires blend shapes.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/TrendingResponse} and HTTP response
     */
    this.getTrendingWithHttpInfo = function(opts) {
      opts = opts || {};
      var postBody = null;


      var pathParams = {
      };
      var queryParams = {
        'stereoscopicType': opts['stereoscopicType'],
        'category': opts['category'],
        'size': opts['size'],
        'minimumWidth': opts['minimumWidth'],
        'pageNum': opts['pageNum'],
        'isFaceFilter': opts['isFaceFilter'],
        'hasBlendShapes': opts['hasBlendShapes'],
        'requiresBlendShapes': opts['requiresBlendShapes'],
      };
      var collectionQueryParams = {
        'type': {
          value: opts['type'],
          collectionFormat: 'csv'
        },
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = ['XAppToken'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = TrendingResponse;

      return this.apiClient.callApi(
        '/vr/trending', 'GET',
        pathParams, queryParams, collectionQueryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType
      );
    }

    /**
     * Trending Endpoint
     * The SVRF Trending Endpoint provides your app or project with the hottest immersive content curated by real humans. The experiences returned mirror the [SVRF homepage](https://www.svrf.com), from timely cultural content to trending pop-culture references. The trending experiences are updated regularly to ensure users always get fresh updates of immersive content.
     * @param {Object} opts Optional parameters
     * @param {Array.<module:model/MediaType>} opts.type The type(s) of Media to be returned (comma separated).
     * @param {module:model/String} opts.stereoscopicType Search only for Media with a particular stereoscopic type.
     * @param {module:model/String} opts.category Search only for Media with a particular category.
     * @param {Number} opts.size The number of results per page. (default to 10)
     * @param {Number} opts.minimumWidth The minimum width for video and photo Media, in pixels.
     * @param {Number} opts.pageNum Pagination control to fetch the next page of results, if applicable.
     * @param {Boolean} opts.isFaceFilter Search only for Face Filters.
     * @param {Boolean} opts.hasBlendShapes Search only for Media that has blend shapes.
     * @param {Boolean} opts.requiresBlendShapes Search only for Media that requires blend shapes.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/TrendingResponse}
     */
    this.getTrending = function(opts) {
      return this.getTrendingWithHttpInfo(opts)
        .then(function(response_and_data) {
          return response_and_data.data;
        });
    }


    /**
     * Search Endpoint
     * The SVRF Search Endpoint brings the power of immersive search found on [SVRF.com](https://www.svrf.com) to your app or project. SVRF&#39;s search engine enables your users to instantly find the immersive experience they&#39;re seeking. Content is sorted by the SVRF rating system, ensuring that the highest quality content and most prevalent search results are returned. 
     * @param {String} q Url-encoded search query.
     * @param {Object} opts Optional parameters
     * @param {Array.<module:model/MediaType>} opts.type The type(s) of Media to be returned (comma separated).
     * @param {module:model/String} opts.stereoscopicType Search only for Media with a particular stereoscopic type.
     * @param {module:model/String} opts.category Search only for Media with a particular category.
     * @param {Number} opts.size The number of results to return per-page, from 1 to 100. (default to 10)
     * @param {Number} opts.minimumWidth The minimum width for video and photo Media, in pixels.
     * @param {Number} opts.pageNum Pagination control to fetch the next page of results, if applicable.
     * @param {Boolean} opts.isFaceFilter Search only for Face Filters.
     * @param {Boolean} opts.hasBlendShapes Search only for Media that has blend shapes.
     * @param {Boolean} opts.requiresBlendShapes Search only for Media that requires blend shapes.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/SearchMediaResponse} and HTTP response
     */
    this.searchWithHttpInfo = function(q, opts) {
      opts = opts || {};
      var postBody = null;

      // verify the required parameter 'q' is set
      if (q === undefined || q === null) {
        throw new Error("Missing the required parameter 'q' when calling search");
      }


      var pathParams = {
      };
      var queryParams = {
        'q': q,
        'stereoscopicType': opts['stereoscopicType'],
        'category': opts['category'],
        'size': opts['size'],
        'minimumWidth': opts['minimumWidth'],
        'pageNum': opts['pageNum'],
        'isFaceFilter': opts['isFaceFilter'],
        'hasBlendShapes': opts['hasBlendShapes'],
        'requiresBlendShapes': opts['requiresBlendShapes'],
      };
      var collectionQueryParams = {
        'type': {
          value: opts['type'],
          collectionFormat: 'csv'
        },
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = ['XAppToken'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = SearchMediaResponse;

      return this.apiClient.callApi(
        '/vr/search', 'GET',
        pathParams, queryParams, collectionQueryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType
      );
    }

    /**
     * Search Endpoint
     * The SVRF Search Endpoint brings the power of immersive search found on [SVRF.com](https://www.svrf.com) to your app or project. SVRF&#39;s search engine enables your users to instantly find the immersive experience they&#39;re seeking. Content is sorted by the SVRF rating system, ensuring that the highest quality content and most prevalent search results are returned. 
     * @param {String} q Url-encoded search query.
     * @param {Object} opts Optional parameters
     * @param {Array.<module:model/MediaType>} opts.type The type(s) of Media to be returned (comma separated).
     * @param {module:model/String} opts.stereoscopicType Search only for Media with a particular stereoscopic type.
     * @param {module:model/String} opts.category Search only for Media with a particular category.
     * @param {Number} opts.size The number of results to return per-page, from 1 to 100. (default to 10)
     * @param {Number} opts.minimumWidth The minimum width for video and photo Media, in pixels.
     * @param {Number} opts.pageNum Pagination control to fetch the next page of results, if applicable.
     * @param {Boolean} opts.isFaceFilter Search only for Face Filters.
     * @param {Boolean} opts.hasBlendShapes Search only for Media that has blend shapes.
     * @param {Boolean} opts.requiresBlendShapes Search only for Media that requires blend shapes.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/SearchMediaResponse}
     */
    this.search = function(q, opts) {
      return this.searchWithHttpInfo(q, opts)
        .then(function(response_and_data) {
          return response_and_data.data;
        });
    }
  };

  return exports;
}));
