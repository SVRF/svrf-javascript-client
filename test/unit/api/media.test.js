import MediaApi from '../../../src/api/media';
import HttpClient from '../../../src/http/http-client';
import QueryService from '../../../src/services/query';

jest.mock('../../../src/http/http-client');

describe('media api', () => {
  let api;
  let httpClient;

  beforeEach(async () => {
    httpClient = new HttpClient();
    api = new MediaApi(httpClient);
  });

  describe('getById', () => {
    it('gets media', async () => {
      const mediaId = 1;
      const mockMedia = {id: mediaId};

      httpClient.get.mockResolvedValue({media: mockMedia});

      const result = await api.getById(mediaId);

      expect(httpClient.get).toHaveBeenCalledWith(`/vr/${mediaId}`);
      expect(result.media).toEqual(mockMedia);
    });

    it('throws an error if http client throws it', async () => {
      const mediaId = 1;

      httpClient.get.mockRejectedValue('error in http request');

      await expect(api.getById(mediaId)).rejects.toThrowErrorMatchingSnapshot();
    });

    it('throws an error if media id is not defined', async () => {
      await expect(api.getById()).rejects.toThrowErrorMatchingSnapshot();
    });

    it('throws an error if media id is empty', async () => {
      await expect(api.getById('')).rejects.toThrowErrorMatchingSnapshot();
    });
  });

  describe('getTrending', () => {
    const mediaList = [{id: 1}, {id: 2}];
    const mockParams = {size: 2};

    beforeAll(() => {
      jest.spyOn(QueryService, 'validateParams').mockReturnValue();
      jest.spyOn(QueryService, 'prepareQueryParams').mockImplementation((params) => ({...params}));
    });

    afterAll(() => {
      QueryService.validateParams.mockRestore();
      QueryService.prepareQueryParams.mockRestore();
    });

    afterEach(() => {
      QueryService.validateParams.mockClear();
      QueryService.prepareQueryParams.mockClear();
    });

    it('gets media if params are not defined', async () => {
      httpClient.get.mockResolvedValue({media: mediaList});

      const result = await api.getTrending();

      expect(result.media).toEqual(mediaList);
    });

    it('gets media if params are defined', async () => {
      httpClient.get.mockResolvedValue({media: mediaList});

      const result = await api.getTrending(mockParams);

      expect(result.media).toEqual(mediaList);
    });

    it('invokes http client get method if params are defined', async () => {
      await api.getTrending(mockParams);

      expect(httpClient.get).toHaveBeenCalledWith('/vr/trending', mockParams);
    });

    it('invokes http client get method if params are not defined', async () => {
      await api.getTrending();

      expect(httpClient.get).toHaveBeenCalledWith('/vr/trending', {});
    });

    it('invokes validator and query service methods if params are defined', async () => {
      await api.getTrending(mockParams);

      expect(QueryService.validateParams).toHaveBeenCalledWith(mockParams);
      expect(QueryService.prepareQueryParams).toHaveBeenCalledWith(mockParams);
    });

    it('invokes validator and query service methods if params are not defined', async () => {
      await api.getTrending();

      expect(QueryService.validateParams).toHaveBeenCalledWith(undefined);
      expect(QueryService.prepareQueryParams).toHaveBeenCalledWith(undefined);
    });

    it('throws an error if http client throws it', async () => {
      httpClient.get.mockRejectedValue(new Error('error in http request'));

      await expect(api.getTrending()).rejects.toThrowErrorMatchingSnapshot();
    });
  });

  describe('search', () => {
    const query = 'query';
    const mediaList = [{id: 1}, {id: 2}];
    const mockParams = {size: 2};

    beforeAll(() => {
      jest.spyOn(QueryService, 'validateParams').mockReturnValue();
      jest.spyOn(QueryService, 'prepareQueryParams').mockImplementation((params) => ({...params}));
    });

    afterAll(() => {
      QueryService.validateParams.mockRestore();
      QueryService.prepareQueryParams.mockRestore();
    });

    afterEach(() => {
      QueryService.validateParams.mockClear();
      QueryService.prepareQueryParams.mockClear();
    });

    it('searches media by query', async () => {
      httpClient.get.mockResolvedValue({media: mediaList});

      const result = await api.search(query, mockParams);

      expect(result.media).toEqual(mediaList);
    });

    it('invokes http client get method if params are defined', async () => {
      await api.search(query, mockParams);

      expect(httpClient.get).toHaveBeenCalledWith('/vr/search', {q: query, ...mockParams});
    });

    it('invokes http client get method if params are not defined', async () => {
      await api.search(query);

      expect(httpClient.get).toHaveBeenCalledWith('/vr/search', {q: query});
    });

    it('invokes validator and query service methods if params are defined', async () => {
      await api.search(query, mockParams);

      expect(QueryService.validateParams).toHaveBeenCalledWith(mockParams);
      expect(QueryService.prepareQueryParams).toHaveBeenCalledWith(mockParams);
    });

    it('invokes validator and query service methods if params are not defined', async () => {
      await api.search(query);

      expect(QueryService.validateParams).toHaveBeenCalledWith(undefined);
      expect(QueryService.prepareQueryParams).toHaveBeenCalledWith(undefined);
    });

    it('throws an error if http client throws it', async () => {
      httpClient.get.mockRejectedValue(new Error('error in http request'));

      await expect(api.search(query)).rejects.toThrowErrorMatchingSnapshot();
    });

    it('throws an error if query is undefined', async () => {
      await expect(api.search()).rejects.toThrowErrorMatchingSnapshot();
    });

    it('throws an error if query is empty', async () => {
      await expect(api.search('')).rejects.toThrowErrorMatchingSnapshot();
    });
  });
});
