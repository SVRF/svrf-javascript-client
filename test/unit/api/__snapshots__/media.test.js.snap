// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`media api getById throws an error if http client throws it 1`] = `undefined`;

exports[`media api getById throws an error if media id is empty 1`] = `"Media Id should be provided"`;

exports[`media api getById throws an error if media id is not defined 1`] = `"Media Id should be provided"`;

exports[`media api getTrending throws an error if http client throws it 1`] = `"error in http request"`;

exports[`media api search throws an error if http client throws it 1`] = `"error in http request"`;

exports[`media api search throws an error if query is empty 1`] = `"query should be provided"`;

exports[`media api search throws an error if query is undefined 1`] = `"query should be provided"`;
